### PR TITLE
feat/improved reachability

### PR DESCRIPTION
**Pull request by SarahU**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
PR changes the words used to indicate the level of Reachability when displaying the data in Json format. It also includes two new classes to remove some functionality from the main command logic.

#### Where should the reviewer start?
format-reachability.ts contains the word changes.
function -> reachable
package -> potentially-reachable
no-info -> no-info (stays the same)

display-test-results.test.ts tests for these conversions using file, test-dep-graph-response-reachable.json, as it contains an example of each of the three states.

vuln-helpers.ts was pulled from test/index.ts - it holds all utility functions for vulnerabilities.
format-test-results.ts was pulled from test/index.ts - it is a collection of functions that handle formatting the output of the Test command.

#### How should this be manually tested?
Run local snyk with test --reachable --json on a relevant project eg https://github.com/snyk123/Semaforo (needs to be a java project).
Run using java 8

#### Any background context you want to provide?
We wanted to update the wording to be consistent with the front end (and as the code was being worked on, did some clean up as well).

#### What are the relevant tickets?
https://snyksec.atlassian.net/jira/software/projects/FLOW/boards/96?selectedIssue=FLOW-433

#### Screenshots
Screenshot 2020-10-14 at 10 17 05

### CODE REVIEW

File: src/cli/commands/test/index.ts [hunk 1]
```diff
@@ -394,56 +366,12 @@ function shouldFail(vulnerableResults: any[], failOn: FailOn) {
...
-}
-
-function isVulnUpgradable(vuln) {
-  return vuln.isUpgradable || vuln.isPinnable;
-}
-
-function isVulnPatchable(vuln) {
-  return vuln.isPatchable;
+function validateSeverityThreshold(severityThreshold) {
+  return SEVERITIES.map((s) => s.verboseName).indexOf(severityThreshold) > -1;
```
Comment (JackuB) [inline src/cli/commands/test/index.ts:370]: Could use `includes` here

```suggestion
  return SEVERITIES.map((s) => s.verboseName).includes(severityThreshold);
```

File: src/cli/commands/test/index.ts [hunk 2]
```diff
@@ -521,6 +449,7 @@ function displayResult(
       : '';
     // user tested a package@version and got 0 vulns back, but there were dev deps
     // to consider
+    // to consider
```
Comment (JackuB) [inline src/cli/commands/test/index.ts:452]: Too much to consider

File: src/cli/commands/test/formatters/format-test-results.ts [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+      dockerSuggestion +=
+        chalk.bold.white(
+          '\n\nPro tip: use `--exclude-base-image-vulns` to exclude from display Docker base image vulnerabilities.',
+        ) + optOutSuggestions;
+    }
+  }
+  return dockerSuggestion;
+}
+
+function groupVulnerabilities(vulns): GroupedVuln[] {
```
Comment (JackuB) [inline src/cli/commands/test/formatters/format-test-results.ts:245]: Maybe I'm misreading, but is it really a type of `GroupedVuln[]`?

Seeing that it's using the `map[curr.id]` I'd say it's not an Array?

**Review by JackuB — APPROVED** [review]

File: src/cli/commands/test/index.ts (hunk 2 above)
Comment (SarahU) [reply src/cli/commands/test/index.ts:452]: lol, indeed.

File: src/cli/commands/test/index.ts (hunk 1 above)
Comment (SarahU) [reply src/cli/commands/test/index.ts:370]: To be honest, I just moved these as is. I shall make a note and can do another change.

File: src/cli/commands/test/formatters/format-test-results.ts (hunk 3 above)
Comment (SarahU) [reply src/cli/commands/test/formatters/format-test-results.ts:245]: Also original code but I will add to my list. Can test it out

**Pull request merged**